include/mgmt/mcumgr/smp_dummy.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Runtime.io 2018. All rights reserved.
+ * Copyright Laird Connectivity 2021-2022. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Dummy transport for the mcumgr SMP protocol for unit testing.
+ */
+#ifndef __SMP_DUMMY_H__
```

Comment:
To align with the existing repository-wide convention, as well as to not use a reserved identifier that starts with `_`

```suggestion
#ifndef ZEPHYR_INCLUDE_MGMT_MCUMGR_SMP_DUMMY_H_
```